dashboard/modules/job/job_manager.py

```diff
@@ -374,60 +406,52 @@ async def run(
-                                "job with SIGTERM."
+                        stop_job_wait_time = int(
+                            os.environ.get(
+                                "RAY_JOB_STOP_WAIT_TIME_S",
+                                self.DEFAULT_RAY_JOB_STOP_WAIT_TIME_S,
                             )
-                            stop_signal = "SIGTERM"
-                        os.killpg(
-                            os.getpgid(child_process.pid),
-                            getattr(signal, stop_signal),
                         )
-                    except ProcessLookupError:
-                        # Process already completed.
+                        poll_job_stop_task = create_task(self._poll_all(child_pid))
+                        await asyncio.wait_for(poll_job_stop_task, stop_job_wait_time)
                         logger.info(
-                            f"Job {self._job_id} completed on its own before it could "
-                            "be manually terminated."
+                            f"Job {self._job_id} has been terminated gracefully "
+                            f"with {stop_signal}."
                         )
-                        pass
-                    else:
-                        # Wait for job to exit gracefully, otherwise kill process
-                        # forcefully after timeout.
-                        try:
-                            stop_job_wait_time = int(
-                                os.environ.get(
-                                    "RAY_JOB_STOP_WAIT_TIME_S",
-                                    self.DEFAULT_RAY_JOB_STOP_WAIT_TIME_S,
-                                )
-                            )
-                            await asyncio.wait_for(polling_task, stop_job_wait_time)
-                            logger.info(
-                                f"Job {self._job_id} has been terminated gracefully "
-                                f"with {stop_signal}."
-                            )
-                        except asyncio.TimeoutError:
-                            logger.warning(
-                                f"Attempt to gracefully terminate job {self._job_id} "
-                                f"through {stop_signal} has timed out after "
-                                f"{stop_job_wait_time} seconds. Job is now being "
-                                "force-killed."
-                            )
-                            polling_task.cancel()
-                            child_process.kill()
+                    except asyncio.TimeoutError:
+                        logger.warning(
+                            f"Attempt to gracefully terminate job {self._job_id} "
+                            f"through {stop_signal} has timed out after "
+                            f"{stop_job_wait_time} seconds. Job is now being "
+                            "force-killed."
+                        )
+                        poll_job_stop_task.cancel()
```

Comment:
This line might not be necessary actually. https://docs.python.org/3/library/asyncio-task.html#asyncio.wait_for 
> If a timeout occurs, it cancels the task and raises [TimeoutError](https://docs.python.org/3/library/exceptions.html#TimeoutError).